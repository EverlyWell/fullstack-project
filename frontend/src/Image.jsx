import React from 'react'

// Redux
import { connect } from 'react-redux'

// Services
import { markFavorite, removeFavorite } from './services/images'

// Libraries
import { makeStyles } from '@material-ui/core/styles'

// Library Components
import Link from '@material-ui/core/Link'
import {
  Favorite as FavoriteIcon,
  FavoriteBorderOutlined as FavoriteOutlinedIcon
} from '@material-ui/icons'


const useStyles = makeStyles((theme) => ({
  container: {
    position: 'relative',
  },
  favorite: {
    position: 'absolute',
    top: theme.spacing(1),
    left: theme.spacing(1),
  },
  favoriteIcon: {
    color: 'white',
  },
  source: {
    position: 'absolute',
    bottom: theme.spacing(1),
    right: theme.spacing(1),
    color: 'white',
  }
}))

function Image({ image, setError, toggleImageFavorite, authToken }) {
  const classes = useStyles()

  function toggleFavorite(event) {
    event.preventDefault()
    setError(null)
    const toggleValue = !image.favorite
    const apiCall = toggleValue ? markFavorite : removeFavorite

    apiCall(authToken, image)
    .then(({ data }) => {
      toggleImageFavorite(image, toggleValue, data.id)
    })
    .catch(({ response }) => {
      setError('Something went wrong. Try again later')
    })
  }

  return (
    <div className={classes.container}>
      <Link
        href='#'
        className={classes.favorite}
        onClick={toggleFavorite}
      >
        {image.favorite
          ? <FavoriteIcon color='secondary' />
          : <FavoriteOutlinedIcon className={classes.favoriteIcon} />
        }
      </Link>

      <img key={image.id} src={image.url} alt='' />

      <Link
        href={image.origin_url}
        target='_blank'
        className={classes.source}
        variant='caption'
      >
        View Source
      </Link>
    </div>
  )
}

const mapStateToProps = ({ authToken }) => {
  return {
    authToken
  }
}
export default connect(
  mapStateToProps,
  null
)(Image)
