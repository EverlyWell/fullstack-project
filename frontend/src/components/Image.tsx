import React, {useContext} from 'react';
import {AppContext} from '../App';

interface ImageInterface {
  id: string;
  url: string;
  title: string;
}

const Image = ({image, favorite}: {image: ImageInterface, favorite: boolean}) => {
  function handleClick(id: string) {
    // TODO
    
  }

  return (
    <div>
      <img src={image.url} alt={image.title} style={{width: '100%'}} />
      <button
        onClick={() => {
          handleClick(image.id);
        }}>
        Favorite
      </button>
    </div>
  );
};

export default Image;
