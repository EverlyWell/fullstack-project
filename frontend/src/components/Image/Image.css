.Image img {
    width: 100%;
    object-fit: contain;
    background: green;
}
