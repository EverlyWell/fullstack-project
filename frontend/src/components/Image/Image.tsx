import React from 'react';

import './Image.css'

export default function Image(props: {imageUrl: string, title: string, className: string}) {
    const { imageUrl, title, className } = props

    return (
        <picture className={className}>
            <source srcSet={imageUrl}/>
            <img alt={title} src={imageUrl}/>
            <p>
                {title}
            </p>
        </picture>
    )
}
