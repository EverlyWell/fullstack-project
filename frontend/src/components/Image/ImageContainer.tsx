import React from 'react'

import Image from './Image'

import { favImage } from '../../services/apiClient'

export default function ImagesContainer(props: {imageUrl: string, title: string, className: string, id: string}) {
    const { imageUrl, title, className, id } = props

    const submitFavImage = (event: any) => {
        favImage(id).then((response) => {
            console.log(response)
        })
    }

    return (
        <Image
            imageUrl={imageUrl}
            title={title}
            className={className}
            onChange={submitFavImage}
        />
    )
}
