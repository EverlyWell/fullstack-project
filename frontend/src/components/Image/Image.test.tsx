import React from 'react';
import { shallow } from 'enzyme';

import Image from '../Image';

describe('Image', () => {
  let tree, props;

  const buildTree = () => {
    const props = {
        imageUrl: 'https://picsum.photos/200/300',
        title: 'photo',
        className: 'Image',
    };

    return shallow(<Image {...props} />);
  };

  it('matches the snapshot', () => {
    tree = buildTree();
    expect(tree).toMatchSnapshot();
  });
});
