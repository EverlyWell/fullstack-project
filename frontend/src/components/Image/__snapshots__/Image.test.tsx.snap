// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches the snapshot 1`] = `
<Image
  className="Image"
  imageUrl="https://picsum.photos/200/300"
  onChange={[Function]}
  title="photo"
/>
`;
